rockcraft/project.py

```diff
@@ -228,6 +228,7 @@ class Project(YamlModel):
     platforms: Dict[str, Any]
     base: Literal["bare", "ubuntu:18.04", "ubuntu:20.04", "ubuntu:22.04"]
     build_base: Optional[Literal["ubuntu:18.04", "ubuntu:20.04", "ubuntu:22.04"]]
+    environment: Optional[Dict[str, Optional[str]]]
```

Comment:
why is the value `Optional[str]` and not `str`? I don't see any examples of that either

Reply:
basically copied from https://github.com/snapcore/snapcraft/blob/main/snapcraft/projects.py#L447. Might be redundant though.

Reply:
@sergiusens given the snapcraft reference, which solution do you prefer? (I'm ok with removing the `Optional`...in fact, that's what I had initially in https://github.com/canonical/rockcraft/commit/c45a50e58c4546e0bcdadb32b58d3d1fe1eae791#diff-0ea7d1a44d76c7792a852505490ca9e3fb738ec2a7814b051ebc828efb8ceceeR229)

Reply:
What's the meaning of the missing string? E.g. what does

```
environment:
  X:
```

mean for the value of `X` in the final container?

Reply:
I tried it out here and apparently the variable get the literal string `"None"` as value

Reply:
alright, I've removed the `Optional`